### PR TITLE
wdp-30

### PR DESCRIPTION
**Pull request by KamiNoSierhej**
https://kindergarten.stxnext.pl/browse/WDP-30

### CODE REVIEW

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                $.getJSON("/api/v1/users", function(result) {
+                    var dropdown = $("#user_id");
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
+                    var selected_user = $("#user_id").val();
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: Declaration

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    var chart_div = $('#chart_div');
+                    if(selected_user) {
+                        loading.show();
+                        chart_div.hide();
+                        $.getJSON("/api/v1/mean_time_weekday/"+selected_user, function(result) {
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: url_for?

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            });
+        })(jQuery);
+    </script>
+
+{% endblock %}
+
+{% block content %}
+
+            <ul>
+                <li><a href="{{ url_for('mainpage') }}">Presence by weekday</a></li>
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: You have to do write loop in base template

File: src/presence_analyzer/templates/mean_time_weekday.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                            var data = new google.visualization.DataTable();
+                            data.addColumn('string', 'Weekday');
+                            data.addColumn('datetime', 'Mean time (h:m:s)');
+                            data.addRows(result);
+                            var options = {
+                                hAxis: {title: 'Weekday'},
+                            },
+                                formatter = new google.visualization.DateFormat({pattern: 'HH:mm:ss'});
+                            formatter.format(data, 1);
+
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/mean_time_weekday.html]: ????

File: src/presence_analyzer/views.py [hunk 5]
```diff
@@ -19,7 +19,23 @@ def mainpage():
     """
     Redirects to front page.
     """
-    return redirect('/static/presence_weekday.html')
+    return render_template('presence_weekday.html')
+
+
+@app.route('/mean_time_view')
+def meantimeview():
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: Write one custom function that gona render all templates

Comment (dpolcyn) [inline src/presence_analyzer/views.py]: What if template doesn't exist? You should handle this exception.

File: src/presence_analyzer/templates/base.html [hunk 6]
```diff
@@ -72,52 +72,25 @@
...
 <body>
     <div id="main">
         <div id="header">
             <h1>Presence analyzer</h1>
-
+{% block content %}
+{% set here = 'main' %}
+{% endblock %}
+        {% set navigate = [
+            (url_for('redirect',where = 'main'),'Presence by weekday','main'),
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/base.html]: Why redirect

File: src/presence_analyzer/templates/base.html [hunk 7]
```diff
@@ -72,52 +72,25 @@
...
+            (url_for('redirect', where = 'mean'),'Presence mean time','mean'),
+            (url_for('redirect', where = 'start-end'),'Presence start-end','startend')
+        ] %}
             <ul>
-                <li id="selected"><a href="/static/presence_weekday.html">Presence by weekday</a></li>
-                <li><a href="/static/mean_time_weekday.html">Presence mean time</a></li>
-                <li><a href="/static/presence_start_end.html">Presence start-end</a></li>
+            {% for href, name, there in navigate %}
+                <li{% if here == there %} id = "selected"{% endif
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/base.html]: style

File: src/presence_analyzer/templates/presence_start_end.html [hunk 8]
```diff
@@ -0,0 +1,62 @@
+{% extends "base.html" %}
+
+{% block scripts %}
+
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_start_end.html]: STYLE

File: src/presence_analyzer/views.py [hunk 9]
```diff
@@ -15,11 +15,19 @@
...
 @app.route('/')
-def mainpage():
+@app.route('/<where>')
+def redirect(where='??'):
     """
-    Redirects to front page.
+    Redirects to pages.
     """
-    return redirect('/static/presence_weekday.html')
+    if where == 'main':
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: It isnt custom function

File: src/presence_analyzer/templates/base.html (hunk 6 above)
Comment (KamiNoSierhej) [inline src/presence_analyzer/templates/base.html]: I named these function redirect because only these funcion doing is 'Redirects to pages.'

File: src/presence_analyzer/views.py [hunk 10]
```diff
@@ -54,12 +63,13 @@ def mean_time_weekday_view(user_id):
...
 
 
-@app.route('/api/v1/presence_weekday/<int:user_id>', methods=['GET'])
+@app.route('/api/v1/presence_weekday/<user_id>', methods=['GET'])
 @jsonify
 def presence_weekday_view(user_id):
     """
     Returns total presence time of given user grouped by weekday.
     """
+    user_id = int(user_id)
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: Why did you add this

File: src/presence_analyzer/templates/base.html [hunk 11]
```diff
@@ -72,52 +72,25 @@
...
+{% block scripts %}{% endblock %}
 <body>
     <div id="main">
         <div id="header">
             <h1>Presence analyzer</h1>
-
+{% block content %}
+{% set here = 'main' %}
+{% endblock %}
+        {% set navigate = [
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/base.html]: Should be on the top of file

File: src/presence_analyzer/views.py (hunk 10 above)
Comment (KamiNoSierhej) [inline src/presence_analyzer/views.py]: user_id is unicode

File: src/presence_analyzer/templates/base.html [hunk 12]
```diff
@@ -72,52 +77,20 @@
...
-
+{% block content %}
+{% set here = 'main' %}
+{% endblock %}
             <ul>
-                <li id="selected"><a href="/static/presence_weekday.html">Presence by weekday</a></li>
-                <li><a href="/static/mean_time_weekday.html">Presence mean time</a></li>
-                <li><a href="/static/presence_start_end.html">Presence start-end</a></li>
+                {% for href, name, there in navigate %}
+                <li{% if here == there %} id = "selected"{% endif
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/base.html]: Style

File: src/presence_analyzer/views.py (hunk 10 above)
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: Wasn't before? What you made? You had to if this changed

Comment (KamiNoSierhej) [inline src/presence_analyzer/views.py]: It have place only when I using url_for in $.getJSON()
